pkg/libvirt/resolve_vm.go

```diff
@@ -99,3 +112,62 @@ func addToNotExistCache(pid uint64) {
 	}
 	cacheNotExist[pid] = false
 }
+
+func getMetadataVMID(vmid string) (string, error) {
+	metadataURI := config.GetLibvirtMetadataURI()
+	token := config.GetLibvirtMetadataToken()
+	domainName := ""
+
+	parts := strings.Split(vmid, "-")
+	if len(parts) < 3 {
+		return "", fmt.Errorf("the vmid provided is not `machine-qemu-*number*-*name*` : %v", vmid)
+	} else {
+		domainName = strings.Join(parts[3:], "-")
+	}
+
+	conn, err := libvirt.NewConnect("qemu:///system")
+	if err != nil {
+		return "", fmt.Errorf("Failed to connect to hypervisor: %v", err)
+	}
+
+	defer conn.Close()
+
+	domains, err := conn.ListAllDomains(libvirt.CONNECT_LIST_DOMAINS_ACTIVE)
+	if err != nil {
+		return "", fmt.Errorf("Failed to list domains: %v", err)
+	}
+
+	defer func() {
+		for _, dom := range domains {
+			dom.Free()
+		}
+	}()
```

Comment:
Doesn't this re-iterate all domains (say 100)  even if we the vm was found in the first iteration in the loop below? 

I think we should simplify the loop below as 

```
for _, dom := range domains {
  name, err := parseDomainForVmID(dom)
  dom.Free()

  if err == nil {
    return name
  }
}

return "", fmt.Errorf("...")
```

Reply:
On my next iteration using the libvirt library from DigitalOcean that Dave suggested, I directly get the metadata once I have the name of the VM so no more iteration !